osu.Game/Overlays/Comments/VotePill.cs

```diff
@@ -0,0 +1,179 @@
+            api.Queue(request);
+        }
+
+        private void onSuccess(CommentBundle response)
+        {
+            isVoted.Value = !isVoted.Value;
+            setCount(response.Comments.First().VotesCount);
+            IsLoading = false;
+        }
+
+        protected override Container CreateBackground() => new Container
+        {
+            RelativeSizeAxes = Axes.Y,
+            AutoSizeAxes = Axes.X,
+            Children = new Drawable[]
+            {
+                borderContainer = new CircularContainer
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Masking = true,
+                    Children = new Drawable[]
+                    {
+                        background = new Box
+                        {
+                            RelativeSizeAxes = Axes.Both
+                        },
+                        hoverLayer = new Box
+                        {
+                            RelativeSizeAxes = Axes.Both,
+                            Alpha = 0
+                        }
+                    }
+                },
+                sideNumber = new SpriteText
+                {
+                    Anchor = Anchor.CentreLeft,
+                    Origin = Anchor.CentreRight,
+                    Text = "+1",
+                    Font = OsuFont.GetFont(size: 14),
+                    Margin = new MarginPadding { Right = 3 },
+                    Alpha = 0,
+                },
+            },
+        };
+
+        protected override Drawable CreateContent() => votesCounter = new OsuSpriteText
+        {
+            Anchor = Anchor.Centre,
+            Origin = Anchor.Centre,
+            Margin = new MarginPadding { Horizontal = 10 },
+            Font = OsuFont.GetFont(size: 14),
+            AlwaysPresent = true,
+        };
+
+        protected override void OnLoadingStart() => onHoverLostAction();
+
+        protected override void OnLoadingFinished()
+        {
+            if (IsHovered)
+                onHoverAction();
+        }
+
+        protected override bool OnHover(HoverEvent e)
+        {
+            onHoverAction();
+            return base.OnHover(e);
+        }
+
+        protected override void OnHoverLost(HoverLostEvent e)
+        {
+            onHoverLostAction();
+            base.OnHoverLost(e);
+        }
+
+        private void onHoverLostAction()
+        {
+            if (isVoted.Value)
+                hoverLayer.Hide();
+            else
+                sideNumber.Hide();
+
+            borderContainer.BorderThickness = 0;
+        }
+
+        private void onHoverAction()
+        {
+            if (!IsLoading)
+            {
+                borderContainer.BorderThickness = 3;
+
+                if (!isVoted.Value)
+                    sideNumber.Show();
+                else
+                    hoverLayer.Show();
+            }
+        }
+
+        private void setCount(int count) => votesCounter.Text = $"+{count}";
```

Comment:
This feels not quite right. I'd ideally like to see a properly-typed `int` property which then auto-updates when the count changes - maybe a case for a `Bindable`?